### DISCUSSION TITLE
[Chapter 12] 스프링 이벤트의 사용

### DISCUSSION THREAD

**Discussion by ahngilwoong** [cowaine/spring-boot-for-msa#192] [Chapter 12. Spring Events] [answered]
스프링 이벤트가 있다는거를 이번 챕터에서 처음 알았습니다.
두 클래스간의 복잡한 연관관계를 끊어내기 위해 실제 현업에서도 사용해볼법한 기능 인 것 같습니다.

Q. 실제로 현업에서 해당 기능을 사용할때 주의해야 할 점이 무엇이 있을지 얘기해보고 싶고(메모리가 터지지는 않을지)
순서가 중요하여 동기밖에 못쓰는 상황은 스프링 이벤트를 사용하면 오히려 독이 될 지에 대해 여쭤보고 싶습니다.

**Comment by JoisFe** [comment, marked as answer]
비동기로 실행하면 스레드 제한이 없으면 메모리가 터지는 경우를 말씀하시는 거라면 스레드 풀로 적절한 설정을 통해 제한을 하면 괜찮지 않을까 생각합니다.

굳이 비동기로 이벤트 처리를 하지 않더라도 이벤트를 사용하는 경우도 클래스와의 결합도를 낮추기 위해 사용하고 있습니다.
